File: 979-di-string-match/di-string-match.java
class Solution {
    public int[] diStringMatch(String s) {
        int n=s.length();
        int arr[]=new int[n+1];
        int iCount=0;
        for(int i=0;i<s.length();i++){
            if(s.charAt(i)=='I'){
                arr[i]=iCount++;
            }else{
                arr[i]=n--;
            }
        }
        arr[s.length()]=iCount;
        return arr;
    }
}

File: 979-di-string-match/di-string-match.py
class Solution(object):
    def diStringMatch(self, s):
        """
        :type s: str
        :rtype: List[int]
        """
        list=[]
        i,j=0,len(s)
        for k in range(len(s)):
            if(s[k]=='I'):
                list.append(i)
                i+=1
            else:
                list.append(j)
                j-=1
        list.append(j)
        return list

        